JIT generated code fails at runtime to call any CUDA function because it rpathes our stubs library.
See https://github.com/pytorch/pytorch/pull/160179

Errors look like
> cutlass_library/source/tools/util/include/cutlass/util/device_memory.h:67  cutlass::device_memory::allocate: cudaMalloc failed: bytes=4096
> terminate called after throwing an instance of 'cutlass::cuda_exception'
>  what():  std::exception


Author: Alexander Grund (TU Dresden)

diff --git a/torch/_inductor/codecache.py b/torch/_inductor/codecache.py
index dc9f5c25365..432b445b4ac 100644
--- a/torch/_inductor/codecache.py
+++ b/torch/_inductor/codecache.py
@@ -2848,9 +2852,13 @@ def _cuda_lib_options() -> list[str]:
     if is_linux():
         _transform_cuda_paths(lpaths)
         for path in lpaths:
+            extra_ldflags.append(f"-L{path}")
             # -rpath ensures the DLL can find its dependencies when loaded, even
             # if the library path is non-standard.
-            extra_ldflags.extend([f"-L{path}", "-Xlinker", f"-rpath={path}"])
+            # But do not add the stubs folder to rpath as the driver is expected to be found at runtime
+            if os.path.basename(path) != "stubs":
+                extra_ldflags.extend(["-Xlinker", f"-rpath={path}"])
+
         extra_ldflags.append("-lcuda")
         extra_ldflags.append("-lcudart")
     else:
